core/app/models/spree/shipment.rb

```diff
@@ -430,7 +430,7 @@ def manifest_restock(item)
     end
 
     def manifest_unstock(item)
-      stock_location.unstock item.variant, item.quantity, self
+      stock_location.unstock(item.variant, item.quantity, self) if item.variant.track_inventory?
```

Comment:
We are skipping creating a stock movement after the checkout when we're not tracking the inventory.